### PR TITLE
Download institute samples

### LINKED ISSUE #5123: Export table with cases info
Hi
 would like to suggest a feature that allows users to export a table containing information for multiple cases in a structured format. The table should include fields such as:

Case
Sample Name
Sequencing Data
Protocols
Each sample should appear as a separate line in the report, designed as a delivery report.

This feature would be incredibly useful, as currently, different institutions use their own coding systems to identify patients. Retrieving such information often requires clicking multiple times, which can be quite time-consuming.

Having a straightforward export option would save significant time and improve efficiency.

Thank you for considering this request, and I look forward to your response.

Best regards,

teresita
**⚠️ Important:** Please avoid sharing any sensitive or personal information, including text or images, in this issue.

**Is your feature request related to a problem in the current program to new available techology or software? Please describe and add links/citations if appropriate.**
A clear and concise description of what the problem is or what the advance you would like us to incorporate is. Ex. when triaging structural variantS, I would like to [...]

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Closes #5123 

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

File: scout/server/blueprints/institutes/controllers.py [hunk 1]
```diff
@@ -461,58 +500,25 @@ def cases(store, request, institute_id):
...
-    data["nr_cases"] = sum(data["status_ncases"].values())
-
-    sanger_ordered_not_validated: Tuple[Dict[str, list]] = get_sanger_unevaluated(
-        store, institute_id, current_user.email
-    )
-    data["sanger_unevaluated"]: Dict[str, list] = sanger_ordered_not_validated[0]
-    data["sanger_validated_by_others"]: Dict[str, list] = sanger_ordered_not_validated[1]
-
-    # local function to add info to case obj
-    def populate_case_obj(case_obj):
```
Comment (northwestwitch) [inline scout/server/blueprints/institutes/controllers.py:474]: Moved outside this function now

File: scout/server/blueprints/institutes/controllers.py [hunk 2]
```diff
@@ -422,28 +424,65 @@ def _sort_cases(data, request, all_cases):
...
+        "Sample Name",
+        "Analysis",
+        "Phenotype",
+        "Sex",
+        "Tissue",
+        "Case Name",
+        "Case ID",
+        "Analysis date",
+        "Case Status",
+        "Track",
```
Comment (northwestwitch) [inline scout/server/blueprints/institutes/controllers.py:440]: What else?

Comment (dnil) [reply scout/server/blueprints/institutes/controllers.py:440]: Default panel is rather useful for a quick glance. Research open seems straightforward. Parenthood confirmed could be good. Rank model versions. Genome build? Sex confirmed? Pred ancestry? Maybe HPO terms?

The list could potentially be long: slightly hard to reach features e.g. from the `delivery_report` or `exe_ver`/`reference_info`, like what pipeline, pipeline version would likely be appreciated. From delivery report or chanjo/2 average coverage...

But all in all, a good start is a good start, and we can always expand!

File: scout/server/blueprints/institutes/controllers.py [hunk 3]
```diff
@@ -422,28 +424,65 @@ def _sort_cases(data, request, all_cases):
...
-        institute_id(str): An _id of an institute
 
-    Returns:
-        data(dict): includes the cases, how many there are and the limit.
-    """
+def cases(store: MongoAdapter, request: request, institute_id: str) -> dict:
+    """Preprocess case objects for the 'cases' view."""
     data = {}
+
+    # Initialize data (institute info, filters, and case counts)
```
Comment (dnil) [inline scout/server/blueprints/institutes/controllers.py:474]: Could we add these comments to the docstring as well; they would make a nice little description of what the function does!

File: scout/server/blueprints/institutes/controllers.py [hunk 4]
```diff
@@ -461,58 +500,25 @@ def cases(store, request, institute_id):
...
-            populate_case_obj(case_obj)
-            nr_cases_showall_statuses += 1
+            populate_case_obj(case_obj, store)
             case_groups[status].append(case_obj)
+            nr_cases_showall_statuses += 1
 
-    # Retrieve cases for the remaining status categories
+    # Fetch additional cases based on filters
+    name_query = request.form
+    limit = int(request.form.get("search_limit", 100))
```
Comment (dnil) [inline scout/server/blueprints/institutes/controllers.py:521]: This looks good, but I have a nagging feeling we had a reason for the old cumbersome construct?

File: scout/server/blueprints/institutes/controllers.py (hunk 1 above)
Comment (dnil) [reply scout/server/blueprints/institutes/controllers.py:474]: Ok, I very much approve of the general change, but only proof-read this cursorily.

File: tests/server/blueprints/institutes/test_institute_views.py [hunk 5]
```diff
@@ -165,6 +165,31 @@ def test_institute_settings(app, user_obj, institute_obj):
         assert updated_institute["clinvar_submitters"] == form_data["clinvar_emails"]
 
 
+def test_cases_export_samples(app, institute_obj):
```
Comment (dnil) [inline tests/server/blueprints/institutes/test_institute_views.py:168]: 🥇

File: tests/server/blueprints/institutes/test_institute_views.py [hunk 6]
```diff
@@ -180,7 +205,7 @@ def test_cases(app, institute_obj):
         # THEN it should return a page
         assert resp.status_code == 200
 
-        # test query passing parameters in seach form
+        # test query passing parameters in search form
```
Comment (dnil) [inline tests/server/blueprints/institutes/test_institute_views.py:208]: ✅

**Review by dnil — APPROVED** [review]
Nice feature!

File: scout/server/blueprints/institutes/controllers.py (hunk 4 above)
Comment (northwestwitch) [reply scout/server/blueprints/institutes/controllers.py]: Reverted!

**Comment by northwestwitch** [comment]
This looks good now both locally and on stage, so I'm going to merge. We're going to re-test it anyway before the upcoming release!

**Pull request merged**